### PR TITLE
support specification, and add circuitbreaker rule and faultdetect rule feature

### LINKED ISSUE #831: SDK和控制面的配置项统一到specification项目中
**What is the feature you want to add?**

**Why do you want to add this feature?**

**How to implement this feature?**

**Additional context**

### PR DESCRIPTION
**Pull request by andrewshan**
**Please provide issue(s) of this PR:**
Fixes #831
Feats #893

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [x] ApiServer
- [x] Auth
- [x] Configuration
- [x] Naming
- [ ] HealthCheck
- [ ] Metrics
- [x] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: apiserver/apiserver.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func GetClientOpenMethod(include []string, protocol string) (map[string]bool, er
 	for _, item := range include {
 		if methods, ok := clientAccess[item]; ok {
 			for _, method := range methods {
-				method = "/v1.Polaris" + strings.ToUpper(protocol) + "/" + method
+				method = "/polaris.v1.Polaris" + strings.ToUpper(protocol) + "/" + method
```
Comment (chuntaojun) [inline apiserver/apiserver.go:96]: 这里 grpc 方跟之前的有出入，是不是还需要兼容下老版本SDK的接入？

Comment (andrewshan) [reply apiserver/apiserver.go]: 是的，之前是因为和现有版本冲突，我改回去/v1

File: apiserver/httpserver/v2/naming_console_access.go [hunk 2]
```diff
@@ -84,21 +88,40 @@ func (h *HTTPServerV2) addDefaultAccess(ws *restful.WebService) {
 	ws.Route(enrichEnableRoutingsApiDocs(ws.PUT("/routings/enable").To(h.EnableRoutings)))
 }
 
+const (
+	deprecatedRoutingV2TypeUrl = "type.googleapis.com/v2."
+	newRoutingV2TypeUrl        = "type.googleapis.com/polaris.v1."
```
Comment (chuntaojun) [inline apiserver/httpserver/v2/naming_console_access.go:93]: 这里应该是 type.googleapis.com/v1. 吧？

**Comment by chuntaojun** [comment]
Screenshot 2023-01-11 at 11 45 16

这里的 typeUrl ，两个保证一样吗？不一样的话，存量会有问题

File: common/model/routing.go [hunk 3]
```diff
@@ -50,50 +50,50 @@ var (
 )
 
 func init() {
-	ruleAny, _ := ptypes.MarshalAny(&apiv2.RuleRoutingConfig{})
-	metaAny, _ := ptypes.MarshalAny(&apiv2.MetadataRoutingConfig{})
```
Comment (chuntaojun) [inline common/model/routing.go:54]: 可能存在存量数据兼容性问题

**Review by chuntaojun — APPROVED** [review]

**Comment by andrewshan** [comment]
> Screenshot 2023-01-11 at 11 45 16
> 
> 这里的 typeUrl ，两个保证一样吗？不一样的话，存量会有问题

这边验证了，存数据库的二进制是不带url的，url是程序在反序列化的时候加入的，反序列化二进制的时候不依赖url，所以存量数据加载不会有问题

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**